extensions/amp-story-shopping/0.1/amp-story-shopping-config.js

```diff
@@ -26,11 +128,21 @@ export let KeyedShoppingConfigDef;
 export function getShoppingConfig(pageElement) {
   const element = pageElement.querySelector('amp-story-shopping-config');
   return getElementConfig(element).then((config) => {
-    //TODO(#36412): Add call to validate config here.
-    return keyByProductTagId(config);
+    config['items'].forEach((item) => validateConfig(item));
+
+    if (essentialFieldsAccum.length > 0) {
+      user().warn(
+        'AMP-STORY-SHOPPING-CONFIG',
+        `Essential fields ${JSON.stringify(
+          essentialFieldsAccum
+        )} are missing. Please add them in the shopping config. See the error messages above for more details.`
+      );
+      return;
```

Comment:
Probably better to return `null` here instead of `undefined` because the former represents an intentional absence of any object value, which is fitting considering that we're deliberately not returning the config. In general, avoid empty return statements within methods that are expected to return a value.